packages/itwinui-react/src/core/InputWithDecorations/InputWithDecorations.tsx

```diff
@@ -0,0 +1,134 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import cx from 'classnames';
+import * as React from 'react';
+import {
+  Box,
+  Icon,
+  InputFlexContainer,
+  useMergedRefs,
+  useSafeContext,
+} from '../utils/index.js';
+import type { PolymorphicForwardRefComponent } from '../utils/index.js';
+import { IconButton } from '../Buttons/index.js';
+import type { InputProps } from '../Input/Input.js';
+
+const InputWithDecorationsContext = React.createContext<
+  React.ComponentProps<typeof InputFlexContainer> | undefined
+>(undefined);
+
+const InputWithDecorationsComponent = React.forwardRef((props, ref) => {
+  const { children, size, isDisabled, ...rest } = props;
+  return (
+    <InputWithDecorationsContext.Provider value={{ size, isDisabled }}>
+      <InputFlexContainer
+        isDisabled={isDisabled}
+        size={size}
+        ref={ref}
+        {...rest}
+      >
+        {children}
+      </InputFlexContainer>
+    </InputWithDecorationsContext.Provider>
+  );
+}) as PolymorphicForwardRefComponent<
+  'div',
+  React.ComponentProps<typeof InputFlexContainer>
+>;
+
+//-------------------------------------------------------------------------------
+
+const InputWithDecorationsInput = React.forwardRef((props, ref) => {
+  const { id: idProp, size, setFocus = false, ...rest } = props;
+  const inputRef = React.useRef<HTMLInputElement>(null);
+  const refs = useMergedRefs<HTMLInputElement>(inputRef, ref);
+  const { size: contextSize, isDisabled } = useSafeContext(
+    InputWithDecorationsContext,
+  );
+
+  React.useEffect(() => {
+    if (inputRef.current && setFocus) {
+      inputRef.current.focus();
```

Comment:
no need for this useEffect, it is buggy, and the `setFocus` prop is completely unnecessary.

for focusing, the user can do it though refs

Reply:
removed from InputWithDecorations.
Do we want to remove from Input too?

Reply:
This is merged in #1406